Add getting started and installation instructions to the welcome page

weiji14 opened the pull request:
Editing the front page to include a one-click Binder link to run the Jupyter notebooks, and steps for people to get a conda virtual environment set up locally for the AGU24 pre-conference workshop.

**Screenshot**

![image](https://github.com/user-attachments/assets/d1696c7d-3a23-48b1-932a-6550f132e857)

TODO:
- [x] Adding quickstart binder button and installation instructions
- [x] Add sphinx-design cards in 3x2 grid layout for the 6 tutorials
- [x] Include thumbnails of the best looking figure in each tutorial (do after all tutorials are ready)
- [ ] etc

References:
- Based on https://github.com/GenericMappingTools/egu22pygmt/pull/10
- Sphinx design cards - https://sphinx-design.readthedocs.io/en/sbt-theme/cards.html

weiji14 committed:
Add getting started and installation instructions to the welcome page

Editing the front page to include a one-click Binder link to run the Jupyter notebooks, and steps for people to get a conda virtual environment set up locally for the AGU24 pre-conference workshop.

Based on EGU22 short course at https://github.com/GenericMappingTools/egu22pygmt/pull/10.

weiji14 committed:
Add sphinx-design cards with tutorial titles, authors and tags

Adding the 6 main tutorials as sphinx-design cards in a 3x2 grid. Each card has the tutorial title, author(s), and tags/badges. The tags hint at which libraries/methods will be covered, and also the datasets that are used.

yvonnefroehlich commented on book/intro.md:
```diff
@@ -1,11 +1,158 @@
+{bdg-secondary-line}`japan_quakes`
+{bdg-info}`pandas`
+{bdg-info}`geopandas`
+{bdg-info-line}`airbnb`
```
Not sure about this; I would prefer to include "choropleth map"
```suggestion
{bdg-info-line}`choropleth map`
```

weiji14 replied:
Yeah, I could use some advice on what tags/badges to add. My current thought process is to include:

- bdg-primary: pygmt or gmt
- bdg-primary-line: GMT module used (e.g. `coast`, `plot`, etc)
- bdg-info: External library used (e.g. `pandas`, `xarray`)
- bdg-secondary-line: GMT remote-dataset used (e.g. `earth_relief`, `earth_age`, etc)

The ``` {bdg-info-line}`airbnb` ``` here was intended to show external non GMT remote-datasets used, but I do agree that having AirBnB here looks weird. We could maybe add some tags for concepts being taught such as `choropleth-map` as you suggested, or `scatterplot`, etc.

Ref: color of badges/tags: https://sphinx-design.readthedocs.io/en/sbt-theme/badges_buttons.html#badges

yvonnefroehlich replied:
Yep, I got your main idea behind the structure of these tags / badges and I definitely like this idea!  And I think mentioning special concept / specific visualization is a good addition.  

Even the file contains "airbnb" currently the column “population” is used for the choropleth-map (there are a lot of different columns / quantities in this dataset).

yvonnefroehlich reviewed (commented):
~So far, I did not figure out how to make a useful code suggestion in case an image is included in this cell of the Jupyter notebook. Thus I just list these two aspects:~

I am very sorry I posted this on the wrong PR, and in this PR I have no permission to delete a review or comment.

weiji14 committed:
Merge branch 'main' into doc/welcome-page

weiji14 committed:
Add thumbnail figures for Tutorials 1, 2, 3, 6

weiji14 committed:
Update links to tutorial notebooks/documents

weiji14 committed:
Update book/intro.md

Co-authored-by: Yvonne Fröhlich <94163266+yvonnefroehlich@users.noreply.github.com>

weiji14 committed:
Refresh badge tags for all tutorials

Sort alphabetically, change `bdg-info` to `bdg-success` for more differentiation (also, unsure why primary and info colour is the same now?), and add more tags based on new tutorial updates.

weiji14 committed:
Merge branch 'main' into doc/welcome-page

weiji14 committed:
Add teaser images for Tutorial 4 and 5

Also choosing other teaser images for Tutorial 1, 2, 3.

weiji14 committed:
Use only bottom part of tut04 figure

Figure is too long, so just getting the bottom part only. Using a 75dpi version.

weiji14 committed:
Adaptive layout to utilize vertical space more efficiently

Remove the h2 heading and do some more sphinx-design grid nesting to squeeze out more vertical space for the thumbnail images. The grids are now adaptive too, and will turn from 3 columns to 1 column on narrow width screens (e.g. mobile devices).

Ref: https://sphinx-design.readthedocs.io/en/sbt-theme/grids.html#nesting-grids.

weiji14 committed:
Change URL to tut01_firstfigure

weiji14 commented:
Ok, spent way too much time on the grid layout, but this PR is finally ready for review! Screenshot of the thumbnail images on the front page:

![image](https://github.com/user-attachments/assets/a522896a-597a-4501-ba28-c21a4b1bbbf5)

Had to do `mamba install 'pydata-sphinx-theme=0.15.3'` to get the left sidebar to hide away properly (xref https://github.com/jupyter-book/jupyter-book/issues/2160). I might pin that dependency later, but otherwise, please review this and I'll try to merge this tomorrow.

yvonnefroehlich commented on book/intro.md:
```diff
@@ -1,11 +1,207 @@
+{bdg-secondary-line}`earth_age`
+:::
+
+:::{grid-item-card} Tutorial 2 - Integration with the Scientific Python Ecosystem: Pandas / GeoPandas
```
```suggestion
:::{grid-item-card} Tutorial 2 - Integration with the scientific Python ecosystem: pandas / GeoPandas
```

jhtong33 replied:
I guess @weiji14 try to maintain consistency across all titles by capitalizing the first letter of each word throughout the page. Maybe it's just to ensure consistency on this page or to align with the style used in EGU22?

weiji14 replied:
I'll take @yvonnefroehlich's suggestions on capitalization, we should try to match with the titles on the left side bar too.

yvonnefroehlich commented on book/intro.md:
```diff
@@ -1,11 +1,207 @@
+::::{grid} 1 1 1 1
+:gutter: 1
+
+:::{grid-item-card} Tutorial 3 - Integration with the Scientific Python Ecosystem: Xarray (grids)
```
```suggestion
:::{grid-item-card} Tutorial 3 - Integration with the scientific Python ecosystem: Xarray (grids)
```

yvonnefroehlich reviewed (commented):
Just a few typo fixes for now - but this looks already very nice @weiji14 🙂! Will give this PR a more detailed review later the day!

yvonnefroehlich reviewed (commented):
Thanks, @weiji14, for working on the introduction page. This looks nice 🙂!

I personally like equal-sized cards more, at least a common horizontal line at which the upper edges of the cards of one row align. But if this is difficult to set up, we can leave the layout as is. Also for the EGU workshop, the height of the cards scales with the size of the images (please see https://www.generic-mapping-tools.org/egu22pygmt/intro.html).

yvonnefroehlich commented on book/intro.md:
```diff
@@ -1,11 +1,207 @@
+{bdg-primary-line}`meca`
+{bdg-primary-line}`project`
+{bdg-primary-line}`text`
+{bdg-primary-line}`xyz2grd`
+{bdg-success}`numpy`
```
```suggestion
{bdg-primary-line}`xyz2grd`
{bdg-secondary-line}`earth_relief`
{bdg-success}`numpy`
```

yvonnefroehlich commented on book/intro.md:
```diff
@@ -1,11 +1,207 @@
+{bdg-primary-line}`plot`
+{bdg-secondary-line}`japan_quakes`
+{bdg-success}`pandas`
+{bdg-success}`geopandas`
+{bdg-success-line}`choropleth map`
+:::
```
```suggestion
{bdg-success}`geopandas`
{bdg-success-line}`choropleth map`
{bdg-success-line}`scatter plot`
:::
```

weiji14 committed:
Apply suggestions from code review

Co-authored-by: Yvonne Fröhlich <94163266+yvonnefroehlich@users.noreply.github.com>

weiji14 commented:
> Thanks, @weiji14, for working on the introduction page. This looks nice 🙂!
> 
> I personally like equal-sized cards more, at least a common horizontal line at which the upper edges of the cards of one row align. But if this is difficult to set up, we can leave the layout as is. Also for the EGU workshop, the height of the cards scales with the size of the images (please see https://www.generic-mapping-tools.org/egu22pygmt/intro.html).

Yeah, so the previous EGU22 short course used `sphinx-panels` which is now deprecated in favour of `sphinx-design` (see https://sphinx-design.readthedocs.io/en/sbt-theme/get_started.html#migrating-from-sphinx-panels), and I couldn't quite work out how to get 2 rows with a range of different heights to look nice (the EGU22 one only had 1 rows which was easier). I did try with equal-height cards per-row, but because each thumbnail figure has very different heights (e.g. Tutorial 4 had an extremely tall figure, compared to Tutorial 5 which is very short), there would be a lot of whitespace that didn't look very nice, so I've adapted the grid layout in commit ecf863f69a57e9b9afa5736bcd7b153a2cb068a7. 

I'll merge this in first, so we at least have the installation instructions on the website (and @maxrjones can send the email to the participants). If anyone has time to alter the thumbnail layout, feel free to do that (but I'm guessing there are more important things to fix before that)!

yvonnefroehlich commented:
> > I personally like equal-sized cards more, at least a common horizontal line at which the upper edges of the cards of one row align. But if this is difficult to set up, we can leave the layout as is. Also for the EGU workshop, the height of the cards scales with the size of the images (please see https://www.generic-mapping-tools.org/egu22pygmt/intro.html).
> 
> Yeah, so the previous EGU22 short course used `sphinx-panels` which is now deprecated in favour of `sphinx-design` (see https://sphinx-design.readthedocs.io/en/sbt-theme/get_started.html#migrating-from-sphinx-panels), and I couldn't quite work out how to get 2 rows with a range of different heights to look nice (the EGU22 one only had 1 rows which was easier). I did try with equal-height cards per-row, but because each thumbnail figure has very different heights (e.g. Tutorial 4 had an extremely tall figure, compared to Tutorial 5 which is very short), there would be a lot of whitespace that didn't look very nice, so I've adapted the grid layout in commit [ecf863f](https://github.com/GenericMappingTools/agu24workshop/commit/ecf863f69a57e9b9afa5736bcd7b153a2cb068a7).

No worries. I already thought based on the comment "spent way too much time on the grid layout" that you already tried different layouts 🙈.
If people really want to work with the materials and play around with the codes, they will download the JNs. So, I do not think the layout of the cards is too important, and we should focus on more relevant aspects first.
